starknet-foundry/crates/cast/src/starknet_commands/account/deploy.rs

```diff
@@ -1,5 +1,114 @@
+use starknet::providers::jsonrpc::HttpTransport;
+use starknet::providers::{JsonRpcClient, Provider};
+use starknet::signers::{LocalWallet, SigningKey};
 
 #[derive(Args, Debug)]
 #[command(about = "Deploy an account to the Starknet")]
-pub struct Deploy {}
+pub struct Deploy {
+    /// Name of the account to be deployed
+    #[clap(short, long)]
+    pub name: String,
+
+    /// Max fee for the transaction
+    #[clap(short, long)]
+    pub max_fee: FieldElement,
+}
+
+pub async fn deploy(
+    provider: &JsonRpcClient<HttpTransport>,
+    path: Utf8PathBuf,
+    name: String,
+    network: &str,
+    max_fee: FieldElement,
+) -> Result<FieldElement> {
+    let network = get_network(network)?.get_value();
+
+    let contents = std::fs::read_to_string(path.clone())?;
+    let mut items: serde_json::Value =
+        serde_json::from_str(&contents).expect("failed to parse json file");
+
+    if items[network].is_null() {
+        bail!("Provided network does not have any accounts defined")
+    }
+    if items[network][&name].is_null() {
+        bail!("Account with provided name does not exist")
+    }
+
+    let private_key = SigningKey::from_secret_scalar(
+        parse_number(
+            items
+                .get(network)
+                .and_then(|network| network.get(&name))
+                .and_then(|name| name.get("private_key"))
+                .and_then(serde_json::Value::as_str)
+                .ok_or_else(|| anyhow!("Couldn't get private key from accounts file"))?,
+        )
+        .expect("Couldn't parse private key"),
+    );
+
+    let factory = OpenZeppelinAccountFactory::new(
+        parse_number(OZ_CLASS_HASH).expect("Couldn't parse OpenZeppelins account class hash"),
+        provider.chain_id().await.expect("Couldn't get chain id"),
+        LocalWallet::from_signing_key(private_key),
+        provider,
+    )
+    .await?;
+
+    let deployment = factory.deploy(
+        parse_number(
+            items
+                .get(network)
+                .and_then(|network| network.get(&name))
+                .and_then(|name| name.get("salt"))
+                .and_then(serde_json::Value::as_str)
+                .ok_or_else(|| anyhow!("Couldn't get salt from accounts file"))?,
+        )
+        .expect("Couldn't parse salt"),
+    );
+    let result = deployment.max_fee(max_fee).send().await;
+
+    match result {
+        Ok(result) => match wait_for_tx(provider, result.transaction_hash).await {
+            Ok(_) => {
+                items[network][&name]["deployed"] = serde_json::Value::from(true);
+                std::fs::write(path, serde_json::to_string_pretty(&items).unwrap())
+                    .expect("Couldn't write to accounts file");
+
+                Ok(result.transaction_hash)
+            }
+            Err(message) => Err(anyhow!(message)),
+        },
+        Err(AccountFactoryError::Provider(error)) => handle_rpc_error(error),
+        _ => Err(anyhow!("Unknown RPC error")),
+    }
+}
```

Comment:
Maybe a `map` and `map_err` could be used here to remove nested match statements?